text/chapter12.md

```diff
@@ -1,469 +1,285 @@
-# Callback Hell
+# Asynchronous Effects
 
 ## Chapter Goals
 
-In this chapter, we will see how the tools we have seen so far - namely monad transformers and applicative functors - can be put to use to solve real-world problems. In particular, we will see how we can solve the problem of _callback hell_.
+This chapter focuses on the `Aff` monad, which is similar to the `Effect` monad, but represents _asynchronous_ side-effects. We'll demonstrate examples of asynchronously interacting with the filesystem and making HTTP requests. We'll also cover how to manage sequential and parallel execution of asynchronous effects.
 
 ## Project Setup
 
-The source code for this chapter can be compiled and run using `spago run`. It is also necessary to install the `request` module using NPM:
+New PureScript libraries introduced in this chapter are:
 
-```text
-npm install
-```
-
-## The Problem
+- `aff` - defines the `Aff` monad.
+- `node-fs-aff` - asynchronous filesystem operations with `Aff`.
+- `affjax` - HTTP requests with AJAX and `Aff`.
+- `parallel` - parallel execution of `Aff`.
 
-Asynchronous code in JavaScript typically uses _callbacks_ to structure program flow. For example, to read text from a file, the preferred approach is to use the `readFile` function and to pass a callback - a function that will be called when the text is available:
+The `affjax` library requires the `xhr2` NPM module, which you need to install by running:
```

Comment:
Perhaps you should clarify that `xhr2` is only needed when running this on Node.js?